components/brave_wallet_ui/common/hooks/use-pending-transaction.ts

```diff
@@ -78,6 +79,8 @@ export const usePendingTransactions = () => {
   ])
   const [baseFeePerGas, setBaseFeePerGas] = React.useState<string>(transactionGasEstimates?.baseFeePerGas || '')
   const [currentTokenAllowance, setCurrentTokenAllowance] = React.useState<string>('')
+  const [isCurrentAllowanceUnlimited, setIsCurrentAllowanceUnlimited] =
```

Comment:
Does this need to be an additional state vs a memo based on `currentTokenAllowance`?

Reply:
I'll refactor this to use less state

Reply:
refactored